strawberryfields/backends/fockbackend/ops.py

```diff
@@ -459,34 +416,16 @@ def phase(theta, trunc):
 
 
 @functools.lru_cache()
-def beamsplitter(t, r, phi, trunc, save=False, directory=None):
+def beamsplitter(t, r, phi, trunc):
     r"""
     The beamsplitter :math:`B(cos^{-1} t, phi)`.
     """
     # pylint: disable=bad-whitespace
-    try:
-        prefac = so.load_bs_factors(trunc, directory)
-    except FileNotFoundError:
-        prefac = so.generate_bs_factors(trunc)
-        if save:
-            so.save_bs_factors(prefac, directory)
-
-    dim_array = np.arange(trunc)
-    N = dim_array.reshape((-1, 1, 1, 1, 1))
-    n = dim_array.reshape((1, -1, 1, 1, 1))
-    M = dim_array.reshape((1, 1, -1, 1, 1))
-    k = dim_array.reshape((1, 1, 1, 1, -1))
-
-    tpwr = M-n+2*k
-    rpwr = n+N-2*k
-
-    T = np.power(t, tpwr) if t != 0 else np.where(tpwr != 0, 0, 1)
-    R = np.power(r, rpwr) if r != 0 else np.where(rpwr != 0, 0, 1)
 
-    BS = np.sum(exp(-1j*(pi+phi)*(n-N)) * T * R * prefac[:trunc,:trunc,:trunc,:trunc,:trunc], axis=-1)
-    BS = BS.swapaxes(0, 1).swapaxes(2, 3)
+    theta = np.arccos(t)
+    BS_tw, _, _ = BSgate(theta, phi, cutoff=trunc)
 
-    return BS
+    return BS_tw.transpose((0,2,1,3))
```

Comment:
This looks great! @thisac, now that the requirements.txt file has been modified to pull in the latest development version of the walrus, I believe this transposition can be removed? @nquesada?

Reply:
Still needs to be there according to @nquesada.

Reply:
Okay, might be worth adding a comment above this line explaining (1) why this is done, and (2) when it should be removed. It can even be added as a `# TODO:` comment, which will then show up in pylint as a reminder.

Reply:
Sounds good! I'll add a `TODO` comment.

Reply:
No, not yet. The lastest version only fixes the convention of the beamsplitter.